### PR TITLE
Bare-bones neural network and data loading example notebook

### PR DESCRIPTION
**Pull request by alexbw**

### CODE REVIEW

File: notebooks/neural_network_and_data_loading.ipynb [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+    {
+      "metadata": {
+        "id": "X2DnZo3iYj18",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "import time\n",
+        "from tqdm import tqdm\n",
```
Comment (hawkinsp) [inline notebooks/neural_network_and_data_loading.ipynb]: You aren't using tqdm. Either use it or remove it.

File: notebooks/neural_network_and_data_loading.ipynb [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+        "  epoch_time = time.time() - start_time\n",
+        "\n",
+        "  train_acc = accuracy(params, train_images, train_labels)\n",
+        "  test_acc = accuracy(params, test_images, test_labels)\n",
+        "  print(\"Epoch {} in {:0.2f} sec\".format(epoch, epoch_time))\n",
+        "  print(\"Training set accuracy {}\".format(train_acc))\n",
+        "  print(\"Test set accuracy {}\".format(test_acc))"
+      ],
+      "execution_count": 0,
+      "outputs": []
```
Comment (hawkinsp) [inline notebooks/neural_network_and_data_loading.ipynb]: The model didn't converge for me, I got:
Epoch 0 in 14.25 sec
Training set accuracy 0.09863333404064178
Test set accuracy 0.0958000048995018
Epoch 1 in 13.68 sec
Training set accuracy 0.11600000411272049
Test set accuracy 0.11710000783205032
Epoch 2 in 12.40 sec
Training set accuracy 0.10001666843891144
Test set accuracy 0.0998000055551529
Epoch 3 in 11.75 sec
Training set accuracy 0.11276666820049286
Test set accuracy 0.11420000344514847
Epoch 4 in 12.25 sec
Training set accuracy 0.09751667082309723
Test set accuracy 0.09740000218153
Epoch 5 in 12.13 sec
Training set accuracy 0.09915000200271606
Test set accuracy 0.10100000351667404
Epoch 6 in 11.80 sec
Training set accuracy 0.09878333657979965
Test set accuracy 0.09550000727176666
Epoch 7 in 11.80 sec
Training set accuracy 0.10064999759197235
Test set accuracy 0.10060000419616699
Epoch 8 in 12.36 sec
Training set accuracy 0.09035000205039978
Test set accuracy 0.08940000087022781
Epoch 9 in 12.74 sec
Training set accuracy 0.10218333452939987
Test set accuracy 0.10110000520944595

which looks like non-convergence. Is there a bug?

Comment (alexbw) [reply notebooks/neural_network_and_data_loading.ipynb]: Sorry, I guess I introduced this bug somewhere along the line. Revised the architecture and now doing better than before the bug.

**Review by hawkinsp — APPROVED** [review]
Looks great!
